core/src/main/java/org/elasticsearch/rest/action/RestActions.java

```diff
@@ -200,53 +195,10 @@ public static QueryBuilder urlParamsToQueryBuilder(RestRequest request) {
         return queryBuilder;
     }
 
-    /**
-     * Get Rest content from either payload or source parameter
-     * @param request Rest request
-     * @return rest content
-     */
-    public static BytesReference getRestContent(RestRequest request) {
-        assert request != null;
-
-        BytesReference content = request.content();
-        if (!request.hasContent()) {
-            String source = request.param("source");
-            if (source != null) {
-                content = new BytesArray(source);
-            }
-        }
-
-        return content;
-    }
-
-    public static QueryBuilder getQueryContent(BytesReference source, IndicesQueriesRegistry indicesQueriesRegistry,
+    public static QueryBuilder getQueryContent(XContentParser requestParser, IndicesQueriesRegistry indicesQueriesRegistry,
```

Comment:
I think *eventually* we can remove this one as well. Because as a followup to #22003 we'll be able to remove `IndicesQueriesRegistry` and `ParseFieldMatcher` is also going to go eventually.